lib/node_modules/@stdlib/array/base/mskfilter-map/docs/types/index.d.ts

```diff
@@ -0,0 +1,171 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Array2D, Collection } from '@stdlib/types/array';
+
+/**
+* Array element indices.
+*/
+type Indices = [ number, number ];
+
+/**
+* Callback invoked for each array element.
+*
+* @returns result
+*/
+type Nullary<U, V> = ( this: V ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @returns result
+*/
+type Unary<T, U, V> = ( this: V, value: T ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @param indices - current array element indices
+* @returns result
+*/
+type Binary<T, U, V> = ( this: V, value: T, indices: Indices ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @param indices - current array element indices
+* @param array - input array
+* @returns result
+*/
+type Ternary<T, U, V> = ( this: V, value: T, indices: Indices, array: Array2D<T> ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @param indices - current array element indices
+* @param array - input array
+* @returns result
+*/
+type Callback<T, U, V> = Nullary<U, V> | Unary<T, U, V> | Binary<T, U, V> | Ternary<T, U, V>;
+
+/**
+* Interface describing the main export.
+*/
+interface Routine {
+	/**
+	* Applies a mask to a provided input array and returns a new array after applying a mapping function.
+	*
+	* @param x - input array
+	* @param mask - mask array
+	* @param clbk - callback to invoke
+	* @param thisArg - execution context
+	* @returns output array
+	*
+	* @example
+	* var x = [ 1, 2, 3, 4 ];
+	*
+	* function scale( val ) {
+	* 	return 10 * val;
+	* }
+	*
+	* var y = mskfilterMap( x, [ 0, 1, 0, 1 ], scale );
+	* // returns [ 20, 40 ]
+	*/
+	<T = unknown, U = unknown, V = unknown>( x: Collection<T>, mask: Collection<T>, clbk: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): Array<U>;
```

Comment:
mask does not need to have the same element type as `x`. In which case, it should be typed separately here and below.